test/blackbox/common/DDSBlackboxTestsTransportSHMUDP.cpp

```diff
@@ -0,0 +1,204 @@
+{
+public:
+
+    void SetUp() override
+    {
+        LibrarySettingsAttributes library_settings;
+        switch (GetParam())
+        {
+            case INTRAPROCESS:
+                library_settings.intraprocess_delivery = IntraprocessDeliveryType::INTRAPROCESS_FULL;
+                xmlparser::XMLProfileManager::library_settings(library_settings);
+                break;
+            case DATASHARING:
+                enable_datasharing = true;
+                break;
+            case TRANSPORT:
+            default:
+                break;
+        }
+    }
+
+    void TearDown() override
+    {
+        LibrarySettingsAttributes library_settings;
+        switch (GetParam())
+        {
+            case INTRAPROCESS:
+                library_settings.intraprocess_delivery = IntraprocessDeliveryType::INTRAPROCESS_OFF;
+                xmlparser::XMLProfileManager::library_settings(library_settings);
+                break;
+            case DATASHARING:
+                enable_datasharing = false;
+                break;
+            case TRANSPORT:
+            default:
+                break;
+        }
+    }
+
+};
+
+void run_parametrized_test(
+        bool reliable_writer,
+        bool reliable_reader)
+{
+    // Set test parameters
+    ReliabilityQosPolicyKind writer_reliability =
+            reliable_writer ? RELIABLE_RELIABILITY_QOS : BEST_EFFORT_RELIABILITY_QOS;
+    ReliabilityQosPolicyKind reader_reliability =
+            reliable_reader ? RELIABLE_RELIABILITY_QOS : BEST_EFFORT_RELIABILITY_QOS;
+
+    // Set up
+    PubSubReader<HelloWorldPubSubType> reader(TEST_TOPIC_NAME);
+    PubSubWriter<HelloWorldPubSubType> writer(TEST_TOPIC_NAME);
+
+    auto sub_shm_descriptor = std::make_shared<eprosima::fastdds::rtps::SharedMemTransportDescriptor>();
+    sub_shm_descriptor->segment_size(2 * 1024 * 1024);
+    std::shared_ptr<eprosima::fastdds::rtps::test_UDPv4TransportDescriptor> sub_udp_descriptor =
+            std::make_shared<eprosima::fastdds::rtps::test_UDPv4TransportDescriptor>();
+
+    reader.disable_builtin_transport()
+            .add_user_transport_to_pparams(sub_shm_descriptor)
+            .add_user_transport_to_pparams(sub_udp_descriptor)
+            .reliability(reader_reliability)
+            .durability_kind(VOLATILE_DURABILITY_QOS)
+            .history_kind(KEEP_ALL_HISTORY_QOS)
+            .init();
+    ASSERT_TRUE(reader.isInitialized());
+
+    auto pub_shm_descriptor = std::make_shared<eprosima::fastdds::rtps::SharedMemTransportDescriptor>();
+    pub_shm_descriptor->segment_size(2 * 1024 * 1024);
+
+    auto pub_udp_descriptor = std::make_shared<eprosima::fastdds::rtps::test_UDPv4TransportDescriptor>();
+    std::atomic<uint32_t> messages_on_odd_port{ 0 };  // Messages corresponding to user data
+    pub_udp_descriptor->locator_filter_ = [&messages_on_odd_port](
+        const eprosima::fastdds::rtps::Locator& destination)
+            {
+                if (destination.port % 2)
```

Comment:
I prefer comparisons to be explicit:

```suggestion
                if (0 != (destination.port % 2) )
```